modules/phase_field/src/auxkernels/EBSDReaderAvgDataAux.C

```diff
@@ -54,7 +57,8 @@ EBSDReaderAvgDataAux::precalculateValue()
 
   // get the data for the grain
   else
-    _value = (*_val)(_ebsd_reader.getAvgData(grain_id));
+    _value = isParamValid("phase") ? (*_val)(_ebsd_reader.getAvgData(_phase, grain_id))
```

Comment:
No. This is not a good pattern. Set `_phase` to `libmesh::invalid_uint`. `isParamValid` does costly string comparisons and map lookups. It should only be done _once_ in the constructor.

Reply:
I tried to set the default value to `libmesh::invalid_uint` as you suggested but a number of the reconstruction and grain tracker tests failed when adding a default value.  Those tests do not require a phase at all, but the default is assigned and trips up on the `if (_phase)` logic statements.  They will only run if `_phase` is an optional parameter without a default value.  Would adding an additional input parameter such as `_consider_phase` be a better solution? This would require changes to `PolycrystalEBSD` as well.

Reply:
No. I suggest not using a default value here (it would look confusing in the docs). In the initializer list have 

```
    _phase(isParamValid("phase") ? getParam<unsigned int>("phase") : libmesh::invalid_uint),
```

and instead of `if (_phase)` do `if (_phase != libmesh::invalid_uint)`. No need to add additional parameters.

Reply:
Sounds good, I will try that.